crypto/provider_conf.c

```diff
@@ -313,3 +313,49 @@ void ossl_provider_add_conf_module(void)
     OSSL_TRACE(CONF, "Adding config module 'providers'\n");
     CONF_module_add("providers", provider_conf_init, NULL);
 }
+
+static CONF *ossl_conf_from_params(OSSL_LIB_CTX *ctx, OSSL_PARAM *params)
+{
+    CONF *my_conf = NCONF_new_ex(ctx, NCONF_default());
+    BIO *conf_bio = BIO_new(BIO_s_mem());
+    int ok = 0;
+    int i;
+
+    if (BIO_printf(conf_bio, "[%s]\nactivate = 1\n", "my_sect") < 0)
+        goto err;
+
+    for (i = 0; params[i].key != NULL; i++) {
+        if (params[i].data_type != OSSL_PARAM_UTF8_STRING)
+            continue;
+        if (BIO_printf(conf_bio, "%s = %s\n", params[i].key, (char *)params[i].data) <= 0)
+            goto err;
+    }
+
+    if (NCONF_load_bio(my_conf, conf_bio, NULL) != 1)
+        goto err;
+
+    ok = 1;
+err:
+    if (ok != 1) {
+        NCONF_free(my_conf);
+        my_conf = NULL;
+    }
+    BIO_free(conf_bio);
+
+    return my_conf;
+}
+
+int OSSL_PROVIDER_activate(OSSL_LIB_CTX *libctx, const char *name,
```

Comment:
The name of this function is quite strange. It is only indirectly related to providers at all. It converts params to a config file and loads it. That might end up loading/configuring some providers or it might not depending on whats in the config file. It's even more indirectly related to activation (since it is perfectly possible to configure a provider without activating it).

Reply:
I enforce activation in the built config. But yes, in general you are correct.

Reply:
Not relevant anymore, resolving